core/store/orm/orm.go

```diff
@@ -583,32 +570,25 @@ func (orm *ORM) CreateTx(
 	return tx, nil
 }
 
-// UpdateTx assigns new EthTx details to a transaction, typically used after a
-// failed Eth transaction attempt
-func (orm *ORM) UpdateTx(
-	tx *models.Tx,
-	ethTx *types.Transaction,
-	from *common.Address,
-	sentAt uint64,
-) error {
-	signedRawTx, err := utils.EncodeTxToHex(ethTx)
-	if err != nil {
-		return errors.Wrap(err, "Update(tx) EncodeTxToHex failed")
-	}
+// AddTxAttempt attaches a new attempt to a Tx, after the attempt has been sent to the chain
+func (orm *ORM) AddTxAttempt(tx *models.Tx, transaction *models.Transaction) (*models.TxAttempt, error) {
+	newTx := transaction.Tx()
 
-	tx.From = *from
-	tx.Nonce = ethTx.Nonce()
-	tx.GasPrice = models.NewBig(ethTx.GasPrice())
-	tx.Hash = ethTx.Hash()
-	tx.SentAt = sentAt
-	tx.SignedRawTx = signedRawTx
-	txAttempt := tx.Attempts[0]
-	txAttempt.Hash = tx.Hash
-	txAttempt.GasPrice = tx.GasPrice
-	txAttempt.SentAt = tx.SentAt
-	txAttempt.SignedRawTx = tx.SignedRawTx
+	tx.From = newTx.From               //*from
```

Comment:
When would `from` ever change? Can we have signers change over multiple attempts?

Reply:
We support multiple accounts, so one attempt could come from account N, a second could come from account N+1 and so on.